### PR TITLE
ipc3: helper: Do not silently accept unrecognized component type in IPC

### PR DESCRIPTION
**Pull request by jsarha**
Return an error code from comp_specific_builder() if the component type is not recognized, instead of just silently ignoring the error.

### CODE REVIEW

File: src/ipc/ipc3/helper.c [hunk 1]
```diff
@@ -285,8 +285,10 @@ static void comp_specific_builder(struct sof_ipc_comp *comp,
 #endif
 		break;
 	default:
+		return -EINVAL;
 		break;
```
Comment (lyakh) [inline src/ipc/ipc3/helper.c:289]: please, remove this `break`

Comment (jsarha) [reply src/ipc/ipc3/helper.c:289]: Argh, my sense for coding style may have been permanently damaged due to MISRA-C work I did it my past life.

**Review by lyakh — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX not expecting ipc3 fix to fail, good to merge ?

**Comment by lrudyX** [comment]
> @wszypelt @lrudyX not expecting ipc3 fix to fail, good to merge ?

There was a problem with CI, several PRs are waiting for rerun.

**Comment by dbaluta** [comment]
@jsarha we get some problems with this patch when testing mixer on IPC3.

Looks like there is no case for mixer and it goes on default -EINVAL value. Please see `comp_specific_builder`.

We get this error:


```
[    10468552.448600] (    10468552.000000) c0 dma-trace             src/trace/dma-trace.c:664  ERROR dtrace_add_event(): number of dropped logs = 13
[    10495936.249596] (       27383.800781) c0 dma-trace             src/ipc/ipc3/helper.c:333  ERROR comp_new(): component type not recognized
[    10496074.582923] (         138.333328) c0 dma-trace             src/ipc/ipc3/helper.c:660  ERROR ipc_comp_new(): component cd = NULL
[    10496211.666251] (         137.083328) c0 dma-trace            src/ipc/ipc3/handler.c:1301 ERROR ipc: pipe 1 comp 0 creation failed -22
```

**Comment by lyakh** [comment]
> @jsarha we get some problems with this patch when testing mixer on IPC3.
> 
> Looks like there is no case for mixer and it goes on default -EINVAL value. Please see `comp_specific_builder`.

@dbaluta would someone be submitting a PR to fix this?

**Comment by dbaluta** [comment]
@lyakh I am looking at this.

**Comment by dbaluta** [comment]
@lyakh sent this for review https://github.com/thesofproject/sof/pull/7963

**Pull request merged**